### PR TITLE
TimeframeForm: Use localDateTime instead of Flatpickr

### LINKED ISSUE #128: Disable autocomplete when adding new timeframe
## Describe the bug

Filling a textual datetime description for Start and End of a new Timeframe results in an autocompleted absolute date. To workaround this I added the Timeframe directly to the database.

## To Reproduce

1.  Create new Timeframe
2. Add `-30 days` for Start
3. Move curser to another input field
4. Start becomes `12/1/2021, 12:00 AM GMT+1`

## Expected behavior
The textual datetime description should remain.

## Your Environment

* Module version: 0.10.0
* Dependent module versions: 
* Icinga Web 2 version and modules (System - About): 2.10.1
* Web browser and version:
* PHP version used (`php --version`): 7.4.3
* Browser: Google Chrome Version 101.0.4951.64 (Official Build) (x86_64)

## Additional context

This may be due to a browser "feature", but perhaps this can be disabled.

### PR DESCRIPTION
**Pull request by jhoxhaa**
fix #128

refs #154
refs #172

### CODE REVIEW

**Comment by lippserd** [comment]
@jhoxhaa Please rebase and resolve merge conflicts.

Does the textual datetime description still work? For example `-1 hour`.

File: application/controllers/TimeframeController.php [hunk 1]
```diff
@@ -27,7 +28,13 @@ public function editAction()
         $this->assertPermission('reporting/timeframes');
         $this->addTitleTab($this->translate('Edit Time Frame'));
 
+        $checkboxValue = 'localDateTime';
+        if (DateTime::createFromFormat('Y-m-d H:i:s', $this->timeframe->getStart()) === false) {
+            $checkboxValue = 'text';
+        }
+
         $values = [
+            'start-end' => $checkboxValue,
```
Comment (nilmerg) [inline application/controllers/TimeframeController.php:37]: I don't quite like that this is part of the controller action. This should be handled inside the form. There you can access the populated values with `getPopulatedValue()`.

File: library/Reporting/Web/Forms/TimeframeForm.php [hunk 2]
```diff
@@ -29,26 +28,33 @@ protected function assemble()
...
+        }
 
-        $this->addDecoratedElement($flatpickr, 'text', 'start', [
-            'required'                    => true,
-            'label'                       => 'Start',
-            'placeholder'                 => 'Select a start date or provide a textual datetime description',
-            'data-flatpickr-default-hour' => '00'
+        $this->addElement($elementType, 'start', [
+            'required'      => true,
+            'label'         => 'Start',
```
Comment (nilmerg) [inline library/Reporting/Web/Forms/TimeframeForm.php:50]: The placeholder is now translated, please do so for labels as well.

**Review by nilmerg — CHANGES_REQUESTED** [review]
With this, a textual input is only allowed for `start` and `end` simultaneously. A fixed datetime for `end` only doesn't make much sense, but for `start`. Consider a report that covers results *since 01-01-2022 00:00:00*. Of course, this is the same as the default timeframe `Current Year`, just expressed differently, but it shows that this example is not unreasonable. I'd opt for both expressions as a fixed date is more comprehensible as such a relative expression: `first day of January this year midnight`. (Especially easier to input, with the help of a widget) Though, wait for @lippserd's opinion before changing this.

What you can do right now:
* Update `ScheduleForm` so that it also uses a `localDateTime` element from now on
* Remove all traces from the `flatpickr` in this module
* Remove the `DecoratedElement` trait

**Comment by jhoxhaa** [comment]
> With this, a textual input is only allowed for `start` and `end` simultaneously. A fixed datetime for `end` only doesn't make much sense, but for `start`. Consider a report that covers results _since 01-01-2022 00:00:00_. Of course, this is the same as the default timeframe `Current Year`, just expressed differently, but it shows that this example is not unreasonable. I'd opt for both expressions as a fixed date is more comprehensible as such a relative expression: `first day of January this year midnight`. (Especially easier to input, with the help of a widget) Though, wait for @lippserd's opinion before changing this.

Do you mean we should use fixed datetime only or that we should be able to use both expressions at the same time? 🤔

**Comment by nilmerg** [comment]
Currently it's only fixed + fixed or relative + relative, what I'd like to have is: fixed + relative

**Comment by yhabteab** [comment]
Once this is ready, please make sure that this issue doesn't occur anymore!

- https://github.com/Icinga/icingaweb2-module-reporting/issues/172

**Pull request closed without merging**